# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/credentialmanagement/nsICredentialChooserService.idl
# Commit: 63d9972bab1f
# Full Hash: 63d9972bab1f267a940ec30c7cea7a4c41e54fa6
# Author: Alexandru Marc <amarc@mozila.com>
# Date: 2025-06-06 08:52:03
# Regressor Bug: 1969017
# File Overlap Count: 2
# Description:
#   Revert "Bug 1969015, 1969017: apply code formatting via Lando" for causing leaks @ mochitest.toml
#   
#   This reverts commit a97cb1d6c08bc38e8644bfc916298b847dbac3a1.
#   
#   Revert "Bug 1969017 - Refactor FedCM to use a managed actor rather than tagging onto WindowGlobalParent - r=anti-tracking-reviewers,emz"
# ==============================================================================

diff -r 801e25cb4bfd -r 63d9972bab1f toolkit/components/credentialmanagement/nsICredentialChooserService.idl
--- a/toolkit/components/credentialmanagement/nsICredentialChooserService.idl	Thu Jun 05 18:51:36 2025 +0000
+++ b/toolkit/components/credentialmanagement/nsICredentialChooserService.idl	Thu Jun 05 21:40:49 2025 +0300
@@ -14,6 +14,20 @@
 
 [scriptable, uuid(673ddc19-03e2-4b30-a868-06297e8fed89)]
 interface nsICredentialChooserService : nsISupports {
+  /**
+   * Show UI to allow the user to make a choice between credentials
+   *
+   * @param browsingContext The top-level browsing context of the context requesting a credential.
+   * @param credentials The options the user should choose from
+   * @param callback A callback to notify when the user makes a choice
+   */
+  void showCredentialChooser(in BrowsingContext browsingContext, in Array<jsval> credentials, in nsICredentialChosenCallback callback);
+  /**
+   * Dismiss the credential chooser UI
+   *
+   * @param browsingContext The top-level browsing context of the context requesting a credential.
+   */
+  void cancelCredentialChooser(in BrowsingContext browsingContext);
 
    /**
    * A service function to help any UI. Fetches and serializes images to